Pass ctx to Validate, ValidateUpdate

ncdc opened the pull request:
Pass ctx to Validate/ValidateUpdate. This is useful so strategies can
access data such as the current user.

ncdc commented:
@smarterclayton @thockin PTAL

vmarmol commented:
Assigning to @smarterclayton, feel free to re-assign.

ncdc commented:
@smarterclayton actually, now that I look at the function signature, since PrepareForCreate/Update doesn't return anything right now, we won't be able to stop a create/update from proceeding when there's a referential integrity issue (i.e. not authorized for Bob to access Alice's ImageStream).

ncdc commented:
Does it make sense to have these methods return a ValidationErrorList?

smarterclayton commented:
Maybe an error.  However, that gets into the question of why wouldn't this then be folded into Validate, where there is just two parts of validation in the strategy - inherent, and referential.

smarterclayton commented:
Having Prepare* return an error seems to just be duplicating Validate*

ncdc commented:
I thought you had said you didn't want to do referential integrity checking in the Validate* methods. I'm quite happy to modify Validate and ValidateUpdate to take in the ctx instead.

smarterclayton commented:
Not the core Validate methods - the Strategy validate methods.


> On Mar 30, 2015, at 1:19 PM, Andy Goldstein <notifications@github.com> wrote:
> 
> I thought you had said you didn't want to do referential integrity checking in the Validate* methods. I'm quite happy to modify Validate and ValidateUpdate to take in the ctx instead.
> 
> —
> Reply to this email directly or view it on GitHub.
>

Commit:
Pass ctx to Validate, ValidateUpdate

Pass ctx to Validate/ValidateUpdate. This is useful so strategies can
access data such as the current user.

ncdc commented:
@smarterclayton updated to pass ctx to the Validate* strategy methods

smarterclayton commented on pkg/registry/controller/rest.go:
```diff
@@ -19,6 +19,8 @@ package controller
 import (
 	"fmt"
 
+	"strconv"
```
Nuke the whitespace above this

ncdc replied:
That's from goimports, but ok

ncdc replied:
Fixed

smarterclayton commented:
After you fix that one line LGTM

smarterclayton commented:
Goimports isn't smart enough to know that "fmt" and "strconv" are in the same group, it just knows that strconv doesn't belong with the deeply nested types.
